packages/react-storage/src/components/StorageBrowser/Controller.tsx

```diff
@@ -57,17 +59,21 @@ export function Controller(): null {
             await getLocationCredentials({ permission, scope }),
           region,
         },
+        options: { refresh },
       }),
     [getLocationCredentials, handleListLocationItems, region]
   );
 
   const loadLocationItems = shouldLoadLocationItems(locationItemsState);
   React.useEffect(() => {
-    if (current && loadLocationItems) {
+    if (location.current && loadLocationItems) {
+      // TODO: update to exhaustive call
+      listLocationItems(location.current);
+    } else if (location.current && history.shouldRefresh) {
       // TODO: update to exhaustive call
-      listLocationItems(current);
+      listLocationItems(location.current, true);
     }
-  }, [current, listLocationItems, loadLocationItems]);
+  }, [location, history.shouldRefresh, listLocationItems, loadLocationItems]);
```

Comment:
I wanted to pass `refresh` to the `options` object from  the `listLocationsItems` call to the `handleListLocationItems` call you set up. Don't know if it's necessary though. I  wasn't sure how to let `handleListLocationItems` know that we needed to refresh the list items list when we navigate back to a previous `NavigateItem` that needed to reload location items, hopefully that makes sense

I think I just need to sync with you on how to call `listLocationsItems` again because when we navigate up to a parent folder from the `Navigate` control, location doesn't change technically and so I don't think this `useEffect` would run. Do you have any thoughts on this?

Reply:
Sounds good, we can sort out any required changes to `loadLocationItems`/`listLocationItems`